libavpipe/src/avpipe_mux.c

```diff
@@ -410,11 +412,19 @@ avpipe_mux(
     xctx_t *xctx)
 {
     int ret = 0;
+    int stream_index;
     AVPacket pkt;
     AVPacket *pkts;
     int *valid_pkts;
     io_mux_ctx_t *in_mux_ctx;
-    int64_t first_pts_array[MAX_STREAMS];
+    // last_pts_array is used to store the last pts of each stream in order to detect surprising
+    // gaps between successive frames
+    int64_t last_pts_array[MAX_STREAMS];
+    // pts_per_frame is used to store the pts delta between successive frames used to stamp the
+    // packets of the mux output
+    int64_t pts_per_frame[MAX_STREAMS];
```

Comment:
This is more an estimated frame duration ts   (`est_frame_duration_ts` maybe?)

Reply:
On the one hand, it's an estimated frame duration, but it is currently enforced, so I think `pts_per_frame` is accurate. Even if its not correct to the source stream in some cases, it is always accurate to the output.